pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -38,18 +38,14 @@ func ValidateCertificateSpec(crt *v1alpha2.CertificateSpec, fldPath *field.Path)
 
 	el = append(el, validateIssuerRef(crt.IssuerRef, fldPath)...)
 
-	if len(crt.CommonName) == 0 && len(crt.DNSNames) == 0 {
-		el = append(el, field.Required(fldPath.Child("dnsNames"), "at least one dnsName is required if commonName is not set"))
+	if len(crt.CommonName) == 0 && len(crt.DNSNames) == 0 && len(crt.URISANs) == 0 {
+		el = append(el, field.Required(fldPath.Child("commonName,dnsNames,uriSANs"), "at least one of commonName, dnsNames, or uriSANs must be set"))
```

Comment:
Just use `fldPath` here instead of globbing together key names - that's what we do elsewhere